packages/lesswrong/server/callbacks/postCallbacks.ts

```diff
@@ -168,14 +162,18 @@ getCollectionHooks("Posts").newAfter.add(async (document: DbPost) => {
 });
 
 getCollectionHooks("Posts").editAsync.add(async function updatedPostMaybeTriggerReview (newPost, oldPost) {
-  // Is this a non-draft post that was awaiting moderator review, which just got
-  // approved?
-  if (!newPost.draft && oldPost.authorIsUnreviewed && !newPost.authorIsUnreviewed) {
-    await postPublishedCallback.runCallbacksAsync([newPost]);
-  }
+  // ignore draft posts
+  if (newPost.draft) return
+
   // Is this a post being undrafted?
-  if (!newPost.draft && oldPost.draft) {
+  if (oldPost.draft) {
     await newDocumentMaybeTriggerReview(newPost)
+  }
+  
+  // if the post author is already approved and the post is getting undrafted,
+  // or the post author is getting approved,
+  // then we consider this "publishing" the post
+  if ((oldPost.draft && !newPost.authorIsUnreviewed) || (oldPost.authorIsUnreviewed && !newPost.authorIsUnreviewed)) {
     await postPublishedCallback.runCallbacksAsync([newPost]);
```

Comment:
I refactored this to make sure `postsNewNotifications()` couldn't be triggered twice by one edit